errors/app.errors.js

```diff
@@ -1,3 +1,20 @@
 exports.wrongPathErrorHandler = (req, res, next) => {
   res.status(404).send({ msg: "not found" });
-};
\ No newline at end of file
+};
+
+exports.postgresErrorHandler = (err, req, res, next) => {
+  // console.log(err.code, "<<< Err code");
+  if (err.code === "22P02" || err.code === "23502" || err.code === "23503") {
+    res.status(400).send({ msg: "bad request" });
+  } else {
+    next(err);
+  }
+};
+
+exports.customErrorHandler = (err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+};
```

Comment:
Good work extracting these - there's an argument that you shouldn't have all of those codes in there yet if you're being really strict with TDD... 

You should have a 500 error handler below these which will be your 'if all else fails' error handler. You don't need to write a test for it though